lib/dsc-lib/src/functions/uri.rs

```diff
@@ -48,19 +49,24 @@ impl Function for Uri {
             return Err(DscError::Parser(t!("functions.uri.invalidRelativeUri").to_string()));
         }
 
-        if let Some(uri_without_slashes) = relative_uri.strip_prefix("//") {
-            let scheme = base.scheme();
-            
-            if uri_without_slashes.contains('/') {
-                return Ok(Value::String(format!("{scheme}:{relative_uri}")));
-            }
-            return Ok(Value::String(format!("{scheme}:{relative_uri}/")));
-        }
-
         let result = base.join(relative_uri)
             .map_err(|e| DscError::Parser(format!("{}: {}", t!("functions.uri.invalidRelativeUri"), e)))?;
         
-        Ok(Value::String(result.to_string()))
+        let final_result = if let Some(url::Host::Ipv6(ipv6_addr)) = result.host() {
+            let segments = ipv6_addr.segments();
+            let expanded = format!("{:04X}:{:04X}:{:04X}:{:04X}:{:04X}:{:04X}:{:04X}:{:04X}",
+                segments[0], segments[1], segments[2], segments[3],
+                segments[4], segments[5], segments[6], segments[7]
+            );
+            
+            let addr = Ipv6Addr::from(segments);
+            let compressed = addr.to_string();
+            result.to_string().replace(&compressed, &expanded)
```

Comment:
I don't think you need to have ipv6 expanded, I think compressed is more desirable.  In this case, I think you could handle it in the tests and it's ok for DSC and .NET to have different representations as the functionality is the same